base/libgit2/diff.jl

```diff
@@ -1,24 +1,29 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+# TODO: make this a general purpose solution
+function Base.cconvert(::Type{Ptr{DiffOptionsStruct}}, pathspecs::AbstractString)
+    cs = Base.cconvert(Cstring, pathspecs)
+    data = [Base.unsafe_convert(Cstring, cs)]
+    sa = StrArrayStruct(pointer(data), 1)
+    do_ref = Ref(DiffOptions(pathspec = sa))
+    return do_ref, data, cs
+end
+function Base.unsafe_convert(::Type{Ptr{DiffOptionsStruct}}, tup::Tuple)
+    Base.unsafe_convert(Ptr{DiffOptionStruct}, first(tup))
+end
+
+
+
+
```

Comment:
Really pedantic point, but perhaps 4 line breaks is an unnecessary amount of vertical space?